src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -32,12 +38,14 @@ public void onCompletion(RecordMetadata md, Exception e) {
 			try {
 				if ( LOGGER.isDebugEnabled()) {
 					LOGGER.debug("->  topic:" + md.topic() + ", partition:" +md.partition() + ", offset:" + md.offset());
-					LOGGER.debug("   " + event.toJSON());
+					LOGGER.debug("   " + this.json);
 					LOGGER.debug("   " + event.getNextBinlogPosition());
 					LOGGER.debug("");
 				}
-				config.setInitialPosition(event.getNextBinlogPosition());
-			} catch (IOException | SQLException e1) {
+				if ( this.lastRowInEvent ) {
+					config.setInitialPosition(event.getNextBinlogPosition());
+				}
```

Comment:
So this was the original reason we did this? I mean, what if it fails at this point...etc ?

Maybe this needs to be mentioned in the docs,  since maxwell may crash at this point and has to repeat all these rows.